### PR TITLE
Initialize repository (nodegit)

### PR DESCRIPTION
**Pull request by KaKi87**
The purpose of this PR is to replace the old simple-git/localStorage code by the new nodegit/sqlite methods to perforn a repository initialization using NodeGIt's API and the addRepository mixin.

### CODE REVIEW

**Comment by mittalyashu** [comment]
😅

**Comment by mittalyashu** [comment]
Initializing repository feature includes many sub-features:

1. Block user to open workspace page if repository is not initialized
2. User should still be able to access settings page
3. Update repository data in database
4. Show repository git init status on welcome page
5. Create a app settings for auto-initialise git repository or manual process

### FAQ

**Which case can you have a local and saved non-initialized repository?**

Allowing users to add non-git repository and later adding option to initialize git repository directly from thermal app is an handy feature.

  It is for those situations where the user just wants to add a project to Thermal but doesn't want to initialise the repository.

  Also it will give user more control over how they want to initialize the repository.

**Comment by KaKi87** [comment]
- [ ] Block user to open workspace page if repository is not initialized
- [x] User should still be able to access settings page
- [x] Update repository data in database
- [ ] Show repository git init status on welcome page
- [ ] Create a app settings for auto-initialise git repository or manual process

File: src/renderer/mixins/addRepository.js [hunk 1]
```diff
@@ -122,8 +122,8 @@ export default {
 					$repositoryName: data.name,
 					$directoryName: data.name,
 					$source: "local",
-					$commitFeature: 1,
-					$remoteFeature: 1,
+					$commitFeature: data.isGitRepo,
+					$remoteFeature: data.isGitRepo,
```
Comment (mittalyashu) [inline src/renderer/mixins/addRepository.js]: @KaKi87 Can you tell me more about this `data` object.

**Comment by mittalyashu** [comment]
> Block user to open workspace page if repository is not initialized

How about we hide the sidebar if the directory is not a git repository?

File: src/renderer/mixins/addRepository.js (hunk 1 above)
Comment (mittalyashu) [reply src/renderer/mixins/addRepository.js]: Never mind! After reading the commit message, it makes sense.

😅

**Comment by KaKi87** [comment]
That's what I wanted to do next. Unfortunately, I didn't found out how to do that considering the current UI structure.

**Comment by mittalyashu** [comment]
Yeah, I am also not sure hiding the sidebar for non-git repository is the right choice.

**Comment by mittalyashu** [comment]
> Create an app setting for auto-initialize git repository or manual process

This PR will cover this feature https://github.com/gitthermal/thermal/pull/3596

**Comment by KaKi87** [comment]
What do you think is the right choice ?

**Comment by mittalyashu** [comment]
About what?

**Comment by KaKi87** [comment]
> Yeah, I am also not sure hiding the sidebar for non-git repository is the right choice.

**Comment by mittalyashu** [comment]
Even if we try to disable each item individually in the sidebar that will be a lot of work.

So, to make the process easier, the only possible option is left to hide the sidebar.

**Comment by KaKi87** [comment]
> Even if we try to disable each item individually in the sidebar that will be a lot of work.

I'd like to point out that this is proof that there's a design issue in the DOM structure, it should be simpler.

**Comment by mittalyashu** [comment]
> I'd like to point out that this is proof that there's a design issue in the DOM structure, it should be simpler.

What do you mean? Can you elaborate it with some examples?

**Pull request merged**